tests/test_single_topology.py

```diff
@@ -1772,3 +1772,179 @@ def test_chiral_core_ring_opening():
 
     assert np.sum(chiral_params_1 == 0) == 3
     assert np.sum(chiral_params_1 == DEFAULT_CHIRAL_ATOM_RESTRAINT_K) == 4
+
+
+def permute_atom_indices(mol_a, mol_b, core, seed):
+    """Randomly permute atom indices in mol_a, mol_b independently, and update core"""
+    rng = np.random.default_rng(seed)
+
+    perm_a = rng.permutation(mol_a.GetNumAtoms())
+    perm_b = rng.permutation(mol_b.GetNumAtoms())
+
+    # RenumberAtoms takes inverse permutations
+    # e.g. [3, 2, 0, 1] means atom 3 in the original mol will be atom 0 in the new one
+    inv_perm_a = np.argsort(perm_a)
+    inv_perm_b = np.argsort(perm_b)
+    mol_a = Chem.RenumberAtoms(mol_a, inv_perm_a.tolist())
+    mol_b = Chem.RenumberAtoms(mol_b, inv_perm_b.tolist())
+
+    core = np.array(core)
+    core[:, 0] = perm_a[core[:, 0]]
+    core[:, 1] = perm_b[core[:, 1]]
+
+    return mol_a, mol_b, core
+
+
+@pytest.mark.parametrize("seed", [2024, 2025])
+def test_chiral_core_bond_breaking_raises_error(seed):
+    """Test that we raise assertions for molecules that cannot generate valid dummy-group anchor assignments. In
+    particular, we break two core-core bonds under an identity mapping (with no dummy atoms).
+    """
+    mol_a = Chem.MolFromMolBlock(
+        """
+  Mrv2311 07312412093D
+
+  6  7  0  0  0  0            999 V2000
+    1.7504    0.2003    1.2663 N   0  0  2  0  0  0  0  0  0  0  0  0
+    0.8845   -0.7367    0.7929 O   0  0  0  0  0  0  0  0  0  0  0  0
+    0.0656    0.1573    0.2634 O   0  0  0  0  0  0  0  0  0  0  0  0
+    0.6778    2.1410    1.4645 F   0  0  0  0  0  0  0  0  0  0  0  0
+    0.9965    1.1853    0.5313 C   0  0  1  0  0  0  0  0  0  0  0  0
+    2.3507    0.8788    0.2466 O   0  0  0  0  0  0  0  0  0  0  0  0
+  1  2  1  0  0  0  0
+  5  4  1  0  0  0  0
+  5  1  1  0  0  0  0
+  2  3  1  0  0  0  0
+  6  1  1  0  0  0  0
+  3  5  1  0  0  0  0
+  5  6  1  0  0  0  0
+M  END
+$$$$""",
+        removeHs=False,
+    )
+
+    mol_b = Chem.MolFromMolBlock(
+        """
+  Mrv2311 07312412083D
+
+  6  5  0  0  0  0            999 V2000
+    1.7504    0.2003    1.2663 N   0  0  2  0  0  0  0  0  0  0  0  0
+    0.8845   -0.7367    0.7929 O   0  0  0  0  0  0  0  0  0  0  0  0
+    0.0656    0.1573    0.2634 H   0  0  0  0  0  0  0  0  0  0  0  0
+    0.6778    2.1410    1.4645 F   0  0  0  0  0  0  0  0  0  0  0  0
+    0.9965    1.1853    0.5313 C   0  0  0  0  0  0  0  0  0  0  0  0
+    2.3507    0.8788    0.2466 H   0  0  0  0  0  0  0  0  0  0  0  0
+  1  2  1  0  0  0  0
+  5  4  1  0  0  0  0
+  5  1  1  0  0  0  0
+  6  1  1  0  0  0  0
+  2  3  1  0  0  0  0
+M  END
+$$$$""",
+        removeHs=False,
+    )
+
+    ff = Forcefield.load_default()
+    core = np.array([[0, 0], [1, 1], [2, 2], [3, 3], [4, 4], [5, 5]])
+
+    mol_a, mol_b, core = permute_atom_indices(mol_a, mol_b, core, seed)
```

Comment:
What is the significance of this permutation?

Reply:
This just renumbers atom indices in mol_a and mol_b (and updates the core consistently) to strengthen the test (i.e. make sure the logic tested is independent of atom numbering)